### DISCUSSION TITLE
Windows Development Support

### DISCUSSION THREAD

**Discussion by syke99** [livebud/bud#81] [Ideas]
Native Windows isn't supported yet. You can use Windows Subsystem for Linux (WSL2) for development on Windows machines, but it would be great to support native Windows. Here's a running list of tasks:

- [ ] Running `make install` and other `make` commands fail on Windows machines (details in [this issue](https://github.com/livebud/bud/issues/79))
- [ ] Supporting ExtraFiles ([see comment](https://github.com/livebud/bud/discussions/81#discussioncomment-2804755))
- [ ] v8go Windows support ([see PR](https://github.com/rogchap/v8go/pull/234))
- [ ] You will need a Windows implementation of the `gcc` compiler, as well as access to the `make` command **WITHOUT** installing make with `Chocolatey` (I have steps in a markdown file ready to be added to a PR, I will just need to be able to fork bud again, just like the syscalls fix)

This list will grow over time.

**Comment by matthewmueller** [comment]
Thanks @syke99 for your investigation! I've closed [the previous issue](https://github.com/livebud/bud/issues/7) in favor of this more focused discussion.

**Reply by syke99** [reply]
Definitely!! I'm excited to help get bud fully working on Windows!!

**Comment by matthewmueller** [comment]
Another challenging one is going to be the use of [os/exec.*Cmd.ExtraFiles](https://pkg.go.dev/os/exec#Cmd) which aren't supported on Windows. I use them for the following:

1. Being able to live-reload the server without closing the TCP connection to port 3000
2. Being able to pass live TCP/UDS connections through the binaries to avoid using `time.Sleep(...)` during tests.
3. *(Coming Soon)* Being able to communicate from the application binary to the bud binary to evaluate JS in V8, avoiding us needing to re-compile V8 every file change in development.

**Reply by syke99** [reply]
Ooo yeah, that will be something that will definitely need to be addressed. When I have some time outside of work, I'll do some research into possible workaround/solutions

**Reply by syke99** [reply]
Would the example you provided [here](https://github.com/livebud/bud/pull/75#issuecomment-1133897989), specifically `c.Template.Files["controller/admin/admin.go"] = `... some template ...`, be an example of the functionality of using os/exec.Cmd.ExtraFiles? I've not yet had time to look into a workaround for this, but did see the comment I linked and thought it might be somewhat relavent 😅

**Reply by syke99** [reply]
@matthewmueller after doing some more research on `os/exec.Cmd.ExtraFiles`, I found [this](https://github.com/golang/go/issues/21085#issuecomment-859541729) comment in a discussion on piping file handlers between parent and child processes and it seems support has been added, but the official docs forgot to remove the comment saying ExtraFiles aren't supported on Windows. However, I think my syscall change in package/socket/child.go is actually related to this and proves that it's actually supported (I *think*). I just tested it out using `make example.basic` and it fails saying that it can't find `-lv8`, which is the same error relating to the fact that `v8go` support isn't working on Windows according to the comment above the target `go.build.windows` in the Makefile
![syscall_v8go_error](https://user-images.githubusercontent.com/95064125/169874141-9d5bd4ac-eb38-4b0a-b9e4-afbad51e50a7.png)

**Reply by anborg** [reply]
Does that mean bud fully supported for devs who have windows laptop (no WSL)? If yes, can you please add instructions in Readme.md for windows install. Thanks.

**Reply by matthewmueller** [reply]
Next version of Bud will support windows, as we're removing the V8 dependency. Stay tuned!
